Tests/IntegrationTest/ExchangeControllerTest/ExchangeControllerTests.cs

```diff
@@ -0,0 +1,137 @@
+            Assert.True(response12.StatusCode == System.Net.HttpStatusCode.OK);
+            Assert.True(response13.StatusCode == System.Net.HttpStatusCode.OK);
+            Assert.True(response21.StatusCode == System.Net.HttpStatusCode.OK);
+            Assert.True(response22.StatusCode == System.Net.HttpStatusCode.OK);
+            Assert.True(waitingExchanges1.Count == 1);
+            Assert.True(waitingExchanges2.Count == 1);
+        }
+
+        [Fact]
+        public async Task Exchange_ExchangeConfirm_NonAuthorizedCall()
+        {
+            // Arrange
+            Uri uri = new Uri(TestFixture.BaseUrl, "exchange/ExchangeConfirm/");
+            HttpClient client = TestFixture.CreateClient();
+
+            // Act
+            ExchangeControllerTestsData.ExchangeModel11.StudentId = ExchangeControllerTestsData.StduentGuid.ToString();
+            var jsonModel = JsonConvert.SerializeObject(ExchangeControllerTestsData.ExchangeModel11);
+            StringContent content = new StringContent(jsonModel, System.Text.Encoding.UTF8, "application/json");
+            HttpResponseMessage response = await client.PostAsync(uri, content);
+
+            // Assert
+            Assert.True(response.IsSuccessStatusCode == false);
+            Assert.True(response.StatusCode == System.Net.HttpStatusCode.Unauthorized);
+        }
+
+        [Fact]
+        public async Task Exchange_ExchangeConfirm_UserNotFound()
+        {
+            // Arrange
+            HttpClient client1 = TestFixture.CreateClient();
+            await AuthenticateClient(client1, ExchangeControllerTestsData.Login1);
+            HttpClient client2 = TestFixture.CreateClient();
+            await AuthenticateClient(client2, ExchangeControllerTestsData.Login2);
+
+            // Act
+            var response12 = await SendExchangeRequest(ExchangeControllerTestsData.ExchangeModel12, client1);
+            var response22 = await SendExchangeRequest(ExchangeControllerTestsData.ExchangeModel22, client2);
+
+            // Assert
+            Assert.True(response12.StatusCode == System.Net.HttpStatusCode.OK);
+            Assert.True(response22.StatusCode == System.Net.HttpStatusCode.NotFound);
+        }
+
+
+        private async Task AuthenticateClient(HttpClient client, LoginModel login)
+        {
+            var jsonModel = JsonConvert.SerializeObject(login);
+            StringContent content = new StringContent(jsonModel, Encoding.UTF8, "application/json");
+            HttpResponseMessage response = await client.PostAsync(LoginUri, content);
+
+            string json = await response.Content.ReadAsStringAsync();
+            JObject o = JObject.Parse(json);
+            JToken token = o.SelectToken("$.token");
+            string studentId = (string)o.SelectToken("$.studentId");
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token.ToString());
+            client.DefaultRequestHeaders.Add("StudentId", studentId);
```

Comment:
if you need this line only because of this line "var studentId = client.DefaultRequestHeaders.GetValues("StudentId").FirstOrDefault();" it would be probably better to return studentId as return value (or output parameter)